NDTensors/src/blocksparse/typeparameteraccessors.jl

```diff
@@ -0,0 +1,18 @@
+using .TypeParameterAccessors: TypeParameterAccessors, Position, parameter, parenttype, set_type_parameter
+
+TypeParameterAccessors.position(::Type{<:Block}, ::typeof(Base.ndims)) = Position(1);
+NDTensors.ndims(type::Type{<:Block}) = parameter(type, Base.ndims)
```

Comment:
@mtfishman I wasn't sure if it should move to `TypeParameterAccessors` or just utilize `TypeParameterAccessors`.   Do you think I should move it?

Reply:
What was the purpose of `NDTensors.ndims`, as opposed to just using `Base.ndims`? Was it something about catching an issue with `Base.ndims` on older versions of Julia?

It makes sense to me to move `NDTensors.ndims` and overloads for Base types into `TypeParameterAccessors`, since we might want to use it in the new libraries like `BlockSparseArrays`, etc.

Reply:
@mtfishman I am not totally sure that we do need to have a separate version of `ndims`. Theres a lot of code to look through so I was first trying to keep the current structure relatively the same, remove artifacts that are no longer necessary and make sure all the code goes through the same pipeline. Then in a subsequent sweep we could assess the structure and make modifications (for example we could decide to remove `NDTensors.ndims` and use `Base`)
Regardless, it should be no problem to move it to `TypeParameterAccessors`.

Reply:
Gotcha, makes sense to try to keep the scope of this PR more narrow.